test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -19,13 +87,273 @@
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.pCandidatePairsArray = &( candidatePairArray[ 0 ] );
+    initInfo.pStunBindingRequestTransactionIdStore = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.pStunBindingRequestTransactionIdStore = &( transactionIdStore[ 0 ] );
+    initInfo.cryptoFunctions.randomFxn = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.cryptoFunctions.randomFxn = testRandomFxn;
+    initInfo.cryptoFunctions.crc32Fxn = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.cryptoFunctions.crc32Fxn = testCrc32Fxn;
+    initInfo.cryptoFunctions.hmacFxn = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.cryptoFunctions.hmacFxn = testHmacFxn;
+    initInfo.creds.pLocalUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pLocalUsername = &( buffer[ 0 ] );
+    initInfo.creds.pLocalPassword = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pLocalPassword = &( buffer[ 0 ] );
+    initInfo.creds.pRemoteUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pRemoteUsername = &( buffer[ 0 ] );
+    initInfo.creds.pRemotePassword = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    initInfo.creds.pRemotePassword = &( buffer[ 0 ] );
+    initInfo.creds.pCombinedUsername = NULL;
+
+    result = Ice_Init( &( context ),
+                       &( initInfo ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Init functionality.
+ */
+void test_iceInit( void )
+{
+    IceContext_t context;
+    IceInitInfo_t initInfo = { 0 };
+    IceResult_t result;
+    IceCandidate_t localCandidateArray[ 10 ], remoteCandidateArray[ 10 ];
+    IceCandidatePair_t candidatePairArray[ 100 ];
+    TransactionIdStore_t transactionIdStore[ 16 ];
```

Comment:
Since we are already using these statically allocated arrays for both the tests `test_iceInit_BadParams` and `test_iceInit` , they can be global variables , instead of being declared in the functions

Reply:
Yes but similar situation will come in later api's as well, for a specific api's there could be 3-4 branches and UT, with similar parameters, then the list of Global variables will increase, Also for reading each test the user would have to first go to the global variables lists and find which one are being used this time. Will be less readable in my opinion . 
What do you suggest ?